tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,225 @@
+		// Discuss with the team if we should add a flag to the test to enable the olm deployment and don't do that deployment in different step
+		if skipDeploy {
+			Skip("Skipping the deployment of the operator")
+		}
+
+		When("installed via helm install", func() {
+			BeforeAll(func() {
+				Expect(kubectl.CreateNamespace(namespace)).To(Succeed(), "Namespace failed to be created")
+
+				extraArg := ""
+				if ocp {
+					extraArg = "--set=platform=openshift"
+				}
+
+				Expect(helm.Install("sail-operator", filepath.Join(baseDir, "chart"), "--namespace "+namespace, "--set=image="+image, extraArg)).
+					Should(Succeed(), "Operator failed to be deployed")
+			})
+
+			It("starts successfully", func() {
+				Eventually(kubectl.GetConditions).
+					WithArguments(namespace, "deployment", deploymentName).
+					Should(ContainElement(resourceAvailable), "Operator deployment is not Available; unexpected Condition")
+				Success("Operator deployment is Available")
+
+				Expect(kubectl.GetPodPhase(namespace, "control-plane=sail-operator")).Should(Equal("Running"), "Operator failed to start; unexpected pod Phase")
+				Success("sail-operator pod is Running")
+			})
+
+			It("deploys all the CRDs", func() {
+				Eventually(kubectl.GetCRDs).
+					Should(ContainElements(crds), "Istio CRDs are not present; expected list to contain all elements")
+				Success("Istio CRDs are present")
+			})
+
+			AfterAll(func() {
+				if CurrentSpecReport().Failed() {
+					LogFailure()
+				}
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio resource", func() {
+		for _, version := range istioVersions {
+			// Note: This var version is needed to avoid the closure of the loop
+			version := version
+
+			Context(fmt.Sprintf("version %s", version), func() {
+				BeforeAll(func() {
+					Expect(kubectl.CreateNamespace(controlPlaneNamespace)).To(Succeed(), "Namespace failed to be created")
+				})
+
+				When("the resource is created", func() {
+					BeforeAll(func() {
+						istioYAML := `
+apiVersion: operator.istio.io/v1alpha1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: %s
+  namespace: %s`
+						istioYAML = fmt.Sprintf(istioYAML, version, controlPlaneNamespace)
+						fmt.Printf("Istio CR YAML: %s\n", istioYAML)
+						Expect(kubectl.ApplyString(istioYAML)).
+							Should(Succeed(), "Istio CR failed to be created")
+						Success("Istio CR created")
+					})
+
+					It("updates the Istio resource status to Reconcilied and Ready", func() {
+						Eventually(kubectl.GetConditions).
+							WithArguments(controlPlaneNamespace, "istio", istioName).
+							Should(ContainElement(resourceReconcilied), "Istio is not Reconcilied; unexpected Condition")
+
+						Eventually(kubectl.GetConditions).
+							WithArguments(controlPlaneNamespace, "istio", istioName).
+							Should(ContainElement(resourceReady), "Istio is not Ready; unexpected Condition")
+
+						Success("Istio resource is Reconcilied and Ready")
+					})
```

Comment:
I'd split this into two `It`s, because these are two very different things. Whether the Istio CR is reconciled, depends only on the operator, whereas the Istio CR's readiness status depends on whether Istiod has started properly.

Reply:
Done